backend/routes/middlewares/index.js

```diff
@@ -84,9 +86,41 @@ function checkProject(req, res, next) {
   });
 }
 
+function checkBadState(req, res, next) {
+  const projectId = JSON.parse(req.body.message).projectId;
+
+  // get the latest record
+  Proposal.find({ projectId })
+    .sort({ round: -1 })
+    .limit(1)
+    .exec(async (err, data) => {
+      if (err) {
+        return res.status(400).send(err);
+      }
+      if (!data[0]) {
+        next();
+      }
+      if (data[0].delivered.status !== 2) {
+        const proposalInfo = await getProposalByRecordId(
+          data[0].airtableRecordId
+        );
+        console.log(proposalInfo.fields);
+        if (
+          (proposalInfo.fields["Proposal State"] == "Funded" ||
+            proposalInfo.fields["Proposal State"] == "Granted") && // # maybe check only for funded and NOT granted
+          proposalInfo.fields["Proposal Standing"] != "Completed"
+        ) {
+          return res.status(400).send("Project has an undelivered proposal");
```

Comment:
Function makes sense.
Get top proposal from project, and check if it's funded & completed.